tests/unit/logicblocks/event/processing/services/test_error_handling.py

```diff
@@ -0,0 +1,67 @@
+import asyncio
+from datetime import timedelta
+
+from logicblocks.event.processing.services import ErrorHandlingService
+
+
+class TestErrorHandlingService:
+    async def test_captures_handleable_exceptions_and_continues(self):
+        class TestException(Exception):
+            pass
+
+        invocations = 0
+        handled_errors = 0
+
+        async def invocation_counter() -> None:
+            nonlocal invocations
+            invocations += 1
+            raise TestException()
+
+        def error_handler(ex: BaseException) -> None:
+            if type(ex) is TestException:
+                nonlocal handled_errors
+                handled_errors += 1
+            else:
+                raise ex
+
+        service = ErrorHandlingService(
+            callable=invocation_counter, error_handler=error_handler
+        )
+
+        task = asyncio.create_task(service.execute())
+
+        await asyncio.sleep(timedelta(milliseconds=50).total_seconds())
```

Comment:
I think you can just call `await service.execute()` rather than scheduling a task.